craft_application/services/lifecycle.py

```diff
@@ -194,6 +194,15 @@ def run(self, step_name: str | None, part_names: list[str] | None = None) -> Non
         except Exception as err:  # noqa: BLE001 - Converting general error.
             raise errors.PartsLifecycleError(f"Unknown error: {str(err)}") from err
 
+        if target_step == Step.PRIME:
```

Comment:
I think this will be better integrated with the lifecycle if we use the [callback feature](https://github.com/canonical/craft-parts/blob/main/craft_parts/callbacks.py#L109)
@cmatsuoka what do you think?

Reply:
I do prefer that, but I also want to ensure that `post_prime` is the very last callback we call. There's no lock to prevent you from registering a post-step callback during the lifecycle (in fact, Charmcraft currently does that), and they appear to run in the order in which they were registered.

An ugly workaround for right now could be to register a post-prime step that itself actually registers the real post-prime (it wouldn't technically *guarantee* it, but "very unlikely to happen by accident" is probably a good enough guarantee). But I think the long-term solution for that would be to have craft-parts implement optional callback priorities so we can set this priority to, say, -1000 and then document that if you want your callback to run after this for whatever reason you have to explicitly bump down even lower.

Reply:
I'd use the prime post-step callback to implement post-prime, would that cover all the requirements you have?

Reply:
@cmatsuoka is there a way to guarantee order for the callbacks?

Reply:
Actually... on the other side of the holiday, I can't actually remember why I wanted to guarantee order.